src/main/scala/com/bazaarvoice/sswf/model/ScheduledStep.scala

```diff
@@ -13,4 +17,9 @@ case class DefinedStep[StepEnum <: (Enum[StepEnum] with WorkflowStep)](step: Ste
   stepInput.stepInputString.foreach(input => require(!input.contains("\u0000"), "input may not contain a null byte."))
 
   def withoutResume = this.copy(stepInput = stepInput.copy(resumeProgress = None))
+
+  override def isSameStepAs(other: ScheduledStep[StepEnum]): Boolean = other match {
```

Comment:
isSameStep is not intended to check equality. It compares steps in a specific way( without taking into account transient in progress ) which is not same as equals.